crates/next-custom-transforms/src/transforms/server_actions.rs

```diff
@@ -1926,56 +1986,99 @@ impl<C: Comments> VisitMut for ServerActions<C> {
                                 })),
                             },
                         ));
-                        client_layer_exports
-                            .insert(atom!("default"), (export_expr, ref_id.clone()));
+                        client_layer_exports.insert(
+                            atom!("default"),
+                            (
+                                vec![export_expr],
+                                ModuleExportName::Ident(atom!("default").into()),
+                                ref_id.clone(),
+                            ),
+                        );
                     } else {
-                        let export_expr =
-                            ModuleItem::ModuleDecl(ModuleDecl::ExportDecl(ExportDecl {
+                        let var_name = if in_cache_file {
+                            self.gen_cache_ident()
+                        } else {
+                            self.gen_action_ident()
+                        };
+
+                        let var_ident = Ident::new(var_name.clone(), DUMMY_SP, self.private_ctxt);
+
+                        let export_name_str: Atom = match export_name {
+                            ModuleExportName::Ident(i) => i.sym.clone(),
+                            ModuleExportName::Str(s) => {
+                                Atom::from(s.value.to_string_lossy().as_ref())
+                            }
+                        };
```

Comment:
```suggestion
                        let export_name_str: Wtf8Atom = match export_name {
                            ModuleExportName::Ident(i) => i.sym.clone().into(),
                            ModuleExportName::Str(s) => s.value.clone())
                        };
```
That will later be used to construct a String literal, which is a `Wtf8Atom` anyway.